proxy/proxy.go

```diff
@@ -20,33 +20,43 @@ type ContainerRequest struct {
 }
 
 type Proxy interface {
-	Serve() error
+	Serve(string, bool) error
 	Stop() error
 	GetRequests() ([]ContainerRequest, error)
 }
 
 type HTTPProxy struct {
-	address           string
-	containerRequests []ContainerRequest
-	listener          *net.TCPListener
-	errors            []error
+	address            string
+	containerRequests  []ContainerRequest
+	listener           *net.TCPListener
+	errors             []error
+	imageInspectsCount map[string]int
+	endpointHost       string
+	noBuild            bool
 }
 
 func NewHTTPProxy(listenAt string) *HTTPProxy {
 	return &HTTPProxy{address: listenAt}
 }
 
-func (p *HTTPProxy) Serve() error {
+func (p *HTTPProxy) Serve(endpointHost string, noBuild bool) error {
 	r := mux.NewRouter()
 	r.Path("/v1.18/containers/create").Methods("POST").HandlerFunc(p.create)
 	r.Path("/v1.18/containers/{id}/json").Methods("GET").HandlerFunc(p.inspect)
 	r.Path("/v1.18/containers/{id}/start").Methods("POST").HandlerFunc(p.start)
 	r.Path("/v1.18/containers/json").Methods("GET").HandlerFunc(p.listAll)
 	r.Path("/v1.18/images/{id:.*}/json").Methods("GET").HandlerFunc(p.inspectImage)
+	r.Path("/v1.18/build").Methods("POST").HandlerFunc(p.build)
+	r.Path("/v1.18/images/create").Methods("POST").HandlerFunc(p.createImage)
+	r.Path("/{rest:.*}").HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		log.Infof("Unhandled request to: %s\n\n", r.URL)
+	})
 
 	laddr, _ := net.ResolveTCPAddr("tcp", p.address)
 	listener, _ := net.ListenTCP("tcp", laddr)
 	p.listener = listener
+	p.endpointHost = endpointHost
```

Comment:
This could confuse us or others later. We manipulate the state of the proxy everytime we call serve. I think this is okay, as long as we force every call to serve to supply arguments for the things we are manipulating so that they get reset. I could see a case where we forget to reset something on a subsequent call.